src/main/java/org/cyclonedx/gradle/ScmUrlConverter.java

```diff
@@ -0,0 +1,62 @@
+package org.cyclonedx.gradle;
+
+/**
+ * Convert a Maven SCM URL into a valid HTTPS URL.
+ *
+ * Background:
+ * The Maven XML schema doesn't restrict the SCM URL to be a valid URL but the CycloneDX schema requires a valid URL.
+ * Some popular Java libraries use sometimes the connectionString or the SSH Git URL in the Maven SCM URL element.
+ * Means instead of losing the SCM URL we simply translate the value to an HTTPS URL.
+ */
+public class ScmUrlConverter {
+	private static String GIT_SCHEMA = "git://";
+	private static String GIT_SSH = "git@";
+	private static String HTTPS_SCHEMA = "https://";
+	private static String SCM_PREFIX = "scm:git:";
+
+	/**
+	 * Convert Maven SCM URL to HTTPS URL.
+
+	 * Mappings:

```

Comment:
this mapping is true for github and some others.
not every git provider supports this behavior!!!